### DISCUSSION TITLE
process.env variables are undefined at frontend

### DISCUSSION THREAD

**Discussion by nitz-iron** [vercel/next.js#42664] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.11.0
      npm: 8.19.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.3-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### What browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

next export 

### Describe the Bug

env variables are undefined in the frontend tho I used PUBLIC_NEXT prefix.  
 
question: Can I also use env variables in next export? 

### Expected Behavior

env variables are not undefined in frontend 

### Link to reproduction

https://github.com/nitz-iron/next-env-var-undef

### To Reproduce

1. npm start
2. enter to "/" path
3. in chrome console: process.env is an empty object / process.env.NEXT_PUBLIC_GANAV = undefined

**Comment by oelbaga** [comment]
I cloned your repo, created an environment variable in .env.local called `NEXT_PUBLIC_GANAV=Hello`  and it works. Could there be a mistake in your .env.local file.


<img width="578" alt="Screenshot 2022-11-08 at 12 18 07 PM" src="https://user-images.githubusercontent.com/11394862/200632173-c39a9839-2d95-4c62-a706-255f55b30cc4.png">

**Comment by ghost** [comment]
You're right (I am the author from different user). 
I see it works if I omit the && from the npm start command. 
But what about next export? how can I do it there via npm command?

**Comment by ghost** [comment]
I succeeded to do it like this:

![image](https://user-images.githubusercontent.com/73851139/200648892-e96d5aa2-f624-42ed-94f6-6fbc372d9887.png)

tomorrow I'll check if it works on next served via custom express server, 
pls do not close.

**Comment by icyJoseph** [comment]
Hi,

When passing commands from the CLI, using `&&` effectively separates the CLI instruction into two commands.

```
"start": "NEXT_PUBLIC_GANAV=5 next dev",
```

That's enough. To do the same with export:

```
"export": "NEXT_PUBLIC_FOO=bar next build && next export"
```

**Reply by nitz-iron** [reply]
Hi @icyJoseph, I found out that if I do what you suggest or what I did at the first example in Next served via custom express server it do not work and gives undefined. But if we try what I suggested it works.

**Reply by icyJoseph** [reply]
How do you start your custom server? I am also a bit confused by your terminology, what is it you are trying to actually do? 

My example makes the variable available to the first command. 

A custom server and `next export`? How are you setting these things up?  It doesn't help very much if you asking how to do "A" task, but you are actually trying to do  "B". 

All of this boils down to how shell variables are made available to the commands we execute.
